app/models/concerns/taggable.rb

```diff
@@ -86,12 +88,22 @@ def process_tags(tag_list)
     return unless tag_list.present?
 
     Rails.logger.info "Processing tags: #{tag_list} for record: #{id}"
-    tags = tag_list.compact_blank.join(",")
+    removed_tags = current_tags_list - tag_list
+    tag_list_without_redundant_cognates = tag_list - get_cognates_list(removed_tags)
```

Comment:
Also, if we would like not to call `get_cognates_list` (and query DB_ multiple times, we can first have one single map between tags and their cognates. Lets say `cognates_map`

Then we can do the following:
```ruby
cognates_map.slice(removed_tags).values.flatten
```